### PR TITLE
chore: add edge request id into logger

### PR DESCRIPTION
**Pull request by lzap**
This adds edge request id from headers into logs. This should be according to folks on the forum slack channel the best request id to log along for correlation with the edge API.

Also adds documentation link to the README.

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
Could you pls follow up on what's edge id? is that the edge service, or what edge are we talking and why do we want it's id in logs as "request id"? :open_mouth:

**Comment by lzap** [comment]
Sure, this is an id generated by our edge API (3scale). It appears that the OpenTelemetry effort is not moving forward too much, so I would like our app to write it to logs and also contribute the same into sources/ib too. This will be much easier to get merged, for example the otel PR in sources seems to be stuck.

File: internal/ctxval/edge_id.go
Comment (ezr-ondrej) [inline internal/ctxval/edge_id.go:6]: Could we pls document here that it is coming from edge gateway? :)

**Review by ezr-ondrej — COMMENTED** [review]
So do we want to add it as otel correlation id as well and pass the `X-Rh-Edge-Request-Id` header to requests within our platform?

File: internal/ctxval/edge_id.go
Comment (lzap) [reply internal/ctxval/edge_id.go:6]: Done

**Comment by lzap** [comment]
Yeah I am thinking both, in dev setup it will be obviously empty.

**Comment by ezr-ondrej** [comment]
> Yeah I am thinking both

So it will be done in follow-up PRs? :)
I'm happy to merge this as is if that's the case :+1:

**Comment by ezr-ondrej** [comment]
/retest
well the bot ain't happy for some reason

**Review by ezr-ondrej — APPROVED** [review]

**Comment by lzap** [comment]
> So it will be done in follow-up PRs? :)

What I mean is I would like to log both trace_id (OpenTel id) and request_id (Edge id) in all logs (if available).

But do not merge yet, I forgot to add important part - passing the edge request id to sources/ib HTTP headers! That is primary reason for this patch.

**Comment by lzap** [comment]
Added the header passing.

**Comment by ezr-ondrej** [comment]
Can't we use the edge as request id instead of generating it?

**Comment by lzap** [comment]
Trace id is required and it must have specific properties defined by OpenTelemetry spec. Edge api is generated by 3scale and does not conform to this specs. I refactored the code that if there is no edge API (ephemeral, dev) nothing is logged or passed actually. Would that work? I would really like to have both options for searching.

**Comment by ezr-ondrej** [comment]
> Would that work? I would really like to have both options for searching.

Sure, sure, I :heart: the addition, I just did not understand it and then wanted to make sure we are getting the best out of it :+1:

**Review by ezr-ondrej — APPROVED** [review]
Loving it :+1:
Do we want to add this to Image builder as well? That would be awesome! :orange_heart:

**Comment by lzap** [comment]
That is my plan as a follow up yes.

**Comment by lzap** [comment]
https://github.com/osbuild/image-builder/pull/628

**Pull request merged**
